wbobeirne/joule-extension
# Open new channels

Completes half of #38 & closes #118.

### Description

Adds the ability to open new channels. You specify a connection string, a funding amount, and optionally a push amount and if it should be private.

This also adds an `<AmountField>` component for amount entries, and is used on InvoiceForm. I didn't change it everywhere yet due to keep this small, but I'll follow up with another PR to change them all.

### Steps to Test

1. Click the top right menu and select "Open channel", confirm it opens the modal.
2. Attempt to submit the form, confirm it pops up an error.
2. Attempt to enter an invalid connection string, confirm it highlights the field in red.
3. Enter a valid connection string, confirm it accepts it.
4. Attempt to enter an amount that exceeds your available on-chain balance, confirm it shows an error.
5. Attempt to enter a negative amount, confirm it shows an error.
6. Enter a valid amount, confirm it accepts it.
7. Submit the form, confirm it works (Assuming there are no legitimate errors)
8. Click on "View transaction", confirm you're taken to an explorer with a valid transaction
9. Attempt to submit the same form as before, confirm you get an error about maximum channels

Also make sure the invoice form works still, as its amount field was replaced with the AmountField component created for this.

Note that you won't see the channel in your channel list, since pending channels aren't yet shown (#71) I also didn't include testing instructions for the advanced fields, but my limited testing showed they worked, and you're welcome to give it a shot yourself.

### Screenshots

![screens](https://user-images.githubusercontent.com/649992/51079438-1c2a8e80-1695-11e9-98f4-bbd7566cfc03.png)
